src/transformers/models/llava/processing_llava.py

```diff
@@ -0,0 +1,179 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Processor class for Llava.
+"""
+
+from typing import Callable, List, Optional, Union
+
+from ...feature_extraction_utils import BatchFeature
+from ...processing_utils import ProcessorMixin
+from ...tokenization_utils_base import BatchEncoding, PaddingStrategy, TextInput, TruncationStrategy
+from ...utils import TensorType, is_torch_available
+
+
+if is_torch_available():
+    import torch
+
+IMAGE_TOKEN = "<image>"
+IMAGE_TOKEN_ID = -200
+
+
+class LlavaProcessor(ProcessorMixin):
+    r"""
+    Constructs a Llava processor which wraps a LLama tokenizer and Llava image processor into a single processor.
+
+    [`LlavaProcessor`] offers all the functionalities of [`CLIPImageProcessor`] and [`LlamaTokenizerFast`]. See
+    the docstring of [`~LlavaProcessor.__call__`] and [`~LlavaProcessor.decode`] for more information.
+
+    Args:
+        image_processor (`CLIPImageProcessor`):
+            An instance of [`CLIPImageProcessor`]. The image processor is a required input.
+        tokenizer (`LlamaTokenizerFast`):
+            An instance of [`LlamaTokenizerFast`]. The tokenizer is a required input.
+        image_size (`int`, *optional*, defaults to 224): Image size (assuming a square image)
+    """
+
+    attributes = ["image_processor", "tokenizer"]
+    image_processor_class = "CLIPImageProcessor"
+    tokenizer_class = "LlamaTokenizerFast"
+
+    def __init__(self, image_processor, tokenizer=None, image_size=224, add_end_of_utterance_token=None, **kwargs):
+        if image_processor is None:
+            raise ValueError("You need to specify an `image_processor`.")
+        if tokenizer is None:
+            raise ValueError("You need to specify a `tokenizer`.")
+
+        super().__init__(image_processor, tokenizer)
+        self.current_processor = self.image_processor
+
+    def tokenize_and_fill_token(self, prompt, image_token_index, return_tensors=None):
+        prompt_chunks = [self.tokenizer(chunk).input_ids for chunk in prompt.split("<image>")]
+
+        def insert_separator(X, sep):
+            return [ele for sublist in zip(X, [sep] * len(X)) for ele in sublist][:-1]
+
+        input_ids = []
+        offset = 0
+        if len(prompt_chunks) > 0 and len(prompt_chunks[0]) > 0 and prompt_chunks[0][0] == self.tokenizer.bos_token_id:
+            offset = 1
+            input_ids.append(prompt_chunks[0][0])
+
+        for x in insert_separator(prompt_chunks, [image_token_index] * (offset + 1)):
+            input_ids.extend(x[offset:])
+
+        if return_tensors is not None:
+            if return_tensors == "pt":
+                return torch.tensor(input_ids, dtype=torch.long)
+            raise ValueError(f"Unsupported tensor type: {return_tensors}")
+        return input_ids
+
+    def __call__(
+        self,
+        prompts: Union[List[TextInput], List[List[TextInput]]],
+        images=None,
```

Comment:
cc @ydshieh as can be seen, any multimodal processor has a different signature here, which will make it very difficult to support them on the image-to-text pipeline for instance. Hence we need additional tests that check whether they all comply to the same API. In this case I'd advocate for the `images` keyword argument followed by `text` as that's what most processors use.

Reply:
I would suggest here to use `text` as the argument name to `LlavaProcessor`.

Reply:
Yes, but also reversed order right?

Reply:
Hmm, when I worked on Kosmos-2, I kinda copied from Blip2/InstructBlip, where the order is `images, text`.
I see Fuyu and CLIP processors are  `text, images`.

I don't know if there is a clear criteria to determine this. In this case, we can probably say `images` is the main input, so should be before `text`.

What's your opinion on this order thing?

Reply:
Ok so that's another reason why we need tests for this 😓 perhaps we could extend [test_forward_signature](https://github.com/huggingface/transformers/blob/main/tests/test_modeling_common.py#L523) for both models + processors (currently it's still implemented for text-only models). Personally I would go for `images` and then `text` (and ideally the latter should have been called texts to also be plural).

Reply:
The current `text` is likely because it is the argument name in `PreTrainedTokenizerBase` (so every tokenizer classes)

text: Union[TextInput, PreTokenizedInput, List[TextInput], List[PreTokenizedInput]] = None,